src/test/regress/expected/citus_schema_distribute_undistribute.out

```diff
@@ -0,0 +1,381 @@
+CREATE TABLE tenant1.partition2 PARTITION OF tenant1.partitioned_table FOR VALUES FROM (11) TO (21);
+INSERT INTO tenant1.partitioned_table SELECT i FROM generate_series(1,20) i;
+-- foreign table
+CREATE TABLE tenant1.foreign_table_test (id integer NOT NULL, data text, a bigserial);
+INSERT INTO tenant1.foreign_table_test SELECT i FROM generate_series(1,100) i;
+CREATE EXTENSION postgres_fdw;
+CREATE SERVER foreign_server
+        FOREIGN DATA WRAPPER postgres_fdw
+        OPTIONS (host 'localhost', port :'master_port', dbname 'regression');
+CREATE USER MAPPING FOR CURRENT_USER
+        SERVER foreign_server
+        OPTIONS (user 'postgres');
+CREATE FOREIGN TABLE tenant1.foreign_table (
+        id integer NOT NULL,
+        data text,
+        a bigserial
+)
+        SERVER foreign_server
+        OPTIONS (schema_name 'tenant1', table_name 'foreign_table_test');
+-- create inherited table
+CREATE TABLE tenant1.cities (
+  name       text,
+  population real,
+  elevation  int
+);
+CREATE TABLE tenant1.capitals (
+  state      char(2) UNIQUE NOT NULL
+) INHERITS (tenant1.cities);
+-- create view
+CREATE VIEW tenant1.view1 AS SELECT * FROM tenant1.table1 JOIN tenant1.table2 USING(id);
+WARNING:  "view tenant1.view1" has dependency to "table tenant1.table2" that is not in Citus' metadata
+DETAIL:  "view tenant1.view1" will be created only locally
+HINT:  Distribute "table tenant1.table2" first to distribute "view tenant1.view1"
+-- create materialized view
+CREATE MATERIALIZED VIEW tenant1.view2 AS SELECT * FROM tenant1.table1;
+-- only allowed from coordinator
+SELECT run_command_on_workers($$SELECT citus_schema_distribute('tenant1');$$);
+                       run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,f,"ERROR:  operation is not allowed on this node")
+ (localhost,57638,f,"ERROR:  operation is not allowed on this node")
+(2 rows)
+
+SELECT run_command_on_workers($$SELECT citus_schema_undistribute('tenant1');$$);
+                       run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,f,"ERROR:  operation is not allowed on this node")
+ (localhost,57638,f,"ERROR:  operation is not allowed on this node")
+(2 rows)
+
+-- inherited table not allowed
+SELECT citus_schema_distribute('tenant1');
+ERROR:  tables in a distributed schema cannot inherit or be inherited
+DROP TABLE tenant1.capitals;
+DROP TABLE tenant1.cities;
+-- foreign table not allowed
+SELECT citus_schema_distribute('tenant1');
+ERROR:  cannot create a foreign table in a distributed schema
+ALTER FOREIGN TABLE tenant1.foreign_table SET SCHEMA public;
+-- only allowed for schema owner or superuser
+CREATE USER dummysuper superuser;
+CREATE USER dummyregular;
+SET role 'dummyregular';
+SELECT citus_schema_distribute('tenant1');
+ERROR:  must be owner of schema (null)
```

Comment:
* Could you add a test to show that dummyregular could use citus_schema_distribute() after altering the owner of the schema to dummyregular?
* Could you also add similar tests for undistribute?
* Both after such non-super-user distribute/undistribute tests, it would be nice to query pg_dist_tenant_schema / pg_dist_colocation tables from coordinator / workers.

Reply:
Also, I wonder should we really restrict this to schema owner or should we allow this if the user has some permissions on schema (usage?) (cc: @marcocitus)

Reply:
> or should we allow this if the user has some permissions on schema (usage?) (

probably no